lib/chef_zero/data_store/v1_to_v2_adapter.rb

```diff
@@ -112,9 +112,13 @@ def fix_exceptions
         begin
           yield
         rescue DataAlreadyExistsError => e
-          raise DataAlreadyExistsError.new([ 'organizations', single_org ] + e.path, e)
+          err = DataAlreadyExistsError.new([ 'organizations', single_org ] + e.path, e)
+          err.set_backtrace(e.backtrace)
```

Comment:
Ruby's eventually getting nested exceptions which may be slightly better than doing it this way, but this is a good pattern for when you want to change the class and message of some lower level exception.